File: problem5.c
/*
 * ========================================================================
 *
 *       Filename:  problem5.c
 *
 *    Description:  
 *
 *        Version:  1.0
 *        Created:  Friday, February 01, 2013 03:27:05 HKT
 *       Revision:  none
 *       Compiler:  gcc
 *
 *         Author:  Andy Scout (), andyhuzhill@gmail.com
 *   Organization:  
 *
 * =========================================================================
 */

#include <stdio.h>

int
main(int argc, char *argv[])
{
    long n, i, flag=0;

    n = 2520;
    while(!flag)
    {
        for(i=1; i< 21; i++)
        {
            if (n % i !=0) 
            {
                n ++;
                break;
            }
            else
                if ( i == 20 && n %i == 0) 
                {
                    flag = 1;
                }
        }
    }
    printf ( "%ld\n", n );
    return 0;
}

File: problem5.py
#!/usr/bin/env python
# *-* encoding: utf-8 *-*
#
# =============================================
#      author:    Andy Scout
#    homepage:    http://andyhuzhill.github.com
#
# description:
#
# =============================================



Flag = False

def get():
    global Flag
    n = 2520
    while not Flag:
        for i in range(1, 21):
            if n % i != 0:
                n += 1
                print n
                break
            elif i == 20 and n % i == 0:
                Flag = True
                return n

if __name__ == "__main__":
    res =get()
    print res

